llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2493,11 +2493,13 @@ class VPExtendedReductionRecipe : public VPReductionRecipe {
 /// recipe is abstract and needs to be lowered to concrete recipes before
 /// codegen. The Operands are {ChainOp, VecOp1, VecOp2, [Condition]}.
 class VPMulAccumulateReductionRecipe : public VPReductionRecipe {
-  /// Opcode of the extend recipe.
-  Instruction::CastOps ExtOp;
+  /// Opcodes of the extend recipes.
```

Comment:
nit: It might be clearer to avoid many getters functions suffixed with 0 or 1. One option could be something like:

```c++
struct VecOperandInfo {
  Instruction::CastOps ExtOp{Instruction::CastOps::CastOpsEnd};
  bool IsNonNeg = false;
};

VecOperandInfo VecOpInfo[2];
```

Then instead of getters for each value you could just have `getVecOp0Info()` and `getVecOp1Info()`, which return `VecOperandInfo`.

Reply:
I like that, thanks. Added.